### PR TITLE
Admin merchant functionality

### PR DESCRIPTION
**Pull request by jpc20**
Admins now have all the CRUD functionality that merchants have when interacting with their items.
Added new admin controllers and forms for this functionality.
Visitors can no longer check out their cart without logging in.

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -22,13 +22,9 @@ def create
...
       end
       session.delete(:cart)
-      if current_user
-        current_user.orders << order
-        flash[:success] = "Your order has been created!"
-        redirect_to "/profile/orders"
-      else
-        redirect_to "/orders/#{order.id}"
-      end
+      current_user.orders << order
```
Comment (Ashkanthegreat) [inline app/controllers/orders_controller.rb:25]: Nice cleanup

File: app/controllers/admin/items_controller.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    @item = Item.new
+    @merchant = Merchant.find(params[:merchant_id])
+  end
+
+  def create
+    @merchant = Merchant.find(params[:merchant_id])
+    item = @merchant.items.create(item_params)
+    if item.save
+      flash[:success] = "Your item has been saved"
+      redirect_to admin_merchant_items_path(@merchant.id)
```
Comment (janegreene) [inline app/controllers/admin/items_controller.rb:16]: These paths are intense, nice work!

File: config/routes.rb [hunk 3]
```diff
@@ -11,9 +11,9 @@
   namespace :admin do
     get '/', to: 'dashboard#index'
     resources :users, only: [:index, :show]
-    get '/merchants', to: 'merchants#index'
-    get '/merchants/:id', to: 'merchants#show'
-    patch '/merchants/:id', to: 'merchants#update'
+    resources :merchants, only: [:index, :show, :update] do
+      resources :items
```
Comment (jpc20) [inline config/routes.rb:15]: This allows admins to interact with items with the same CRUD functionality as a mercahnt.

File: app/views/admin/items/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<h1>New Item for <%= @merchant.id %></h1>
+
+<center>
+  <%= form_for [:admin_merchant, @item] do |f|  %>
```
Comment (jpc20) [inline app/views/admin/items/new.html.erb:4]: :admin_merchant routes this form to the admin_merchant path for an item model.

File: app/views/admin/items/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<h1>Edit <%=link_to @item.name, "/items/#{@item.id}" %></h1>
```
Comment (Ashkanthegreat) [inline app/views/admin/items/edit.html.erb:1]: Good use of form_for

File: app/views/admin/items/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    <%= f.label :price %>
+    <%= f.text_field :price %>
+
+    <%= f.label :image %>
+    <%= f.text_field :image %>
+
+    <%= f.label :inventory %>
+    <%= f.text_field :inventory %>
+
```
Comment (Lithnotep) [inline app/views/admin/items/new.html.erb:19]: really getting the hang of these form-fors!

File: spec/features/users/user_admin_spec.rb [hunk 7]
```diff
@@ -151,6 +151,153 @@
...
+    user = create(:user, role: 2)
+    test_item = @meg.items.create(name: "Gatorskins", description: "They'll never pop!", price: 100, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 12)
+    visit "/"
+    click_on "Login"
+    fill_in :email, with: user.email
+    fill_in :password, with: user.password
+    click_on "Log In"
+    visit "/admin/merchants/#{@meg.id}"
+    click_link "Items"
+    within "#item-#{test_item.id}" do
```
Comment (Lithnotep) [inline spec/features/users/user_admin_spec.rb:294]: great job expanding these tests

**Pull request merged**